### ISSUE TITLE
Integrate with OKFN data packages

### ISSUE THREAD

**Issue by richfitz** [traitecoevo/datastorr#2]
Add a packages.json file that contains metadata information probably satisfies most of the requirements.

**Comment by richfitz** [comment]
Here's the website with a bit more information http://data.okfn.org/doc/data-package

Importantly, this can be additional to what we currently have and allow better interopability. I don't believe there is good R tooling for dealing with datapackages yet though.

**Comment by wcornwell** [comment]
So like for `taxonlookup` we take what's now in the github readme.md and put it in a .json file?  I guess ideally it would also be in the R documentation also?  I guess we need a system where the meta-data in one place (the json file?) is canonical and the other 2 are generated?

**Comment by dfalster** [comment]
I really like the idea of the OKFN data packages, so in principle it would be great to support them. Depends how much work it is. Seems low cost.

Generating the readme from a single canonical source for metadata shouldn't be too hard. I tried something like this a while back, where i used a json file with metadata to write the readme. (see readme.Rmd in [github.com/dfalster/Falster_2005_JEcol_data](https://github.com/dfalster/Falster_2005_JEcol_data)). Now I know there is now an actual preferred for that metadata.

**Comment by richfitz** [comment]
Yeah, this is not too much work now that I have the automatic uploading thing worked out.  We'd just hook into the same set of routines.

I think I'd opt to put the json in with the releases themselves, and have the URIs in the release json resolve to the github release URIs.  So for taxonlookup it would read:

``` json
{
  "name" : "traitecoevo/taxonlookup",
  "title" : "A dynamically-updating versioned taxonomic resource for vascular plants",
  "license" : "CC0",
  "sources" : [{
    "name": "The plant list",
    "web": "http://www.theplantlist.org"
  }],
  "author": "Will Cornwell <wcornwell@gmail.com>",
  "contributors": [
    "Will Cornwell <wcornwell@gmail.com>",
    "Rich FitzJohn <rich.fitzjohn@gmail.com>",
    "Matt Pennell <mwpennell@gmail.com>"
  ],
  "version": "1.0.2",
  "resources": [{
    "url": "https://github.com/traitecoevo/taxonlookup/releases/download/v1.0.2/plant_lookup.csv",
    "name": "plant_lookup",
    "format": "csv",
    "hash": "sha1:cf6bb45eed09973d599e97fa8a6b8234c084e52a"
  }]
}
```

as you can see most of that is gettable from the DESCRIPTION file, so that's easy enough.

**Comment by dfalster** [comment]
Looks good.

**Comment by wcornwell** [comment]
I agree, the specific meta-data for the columns might take a bit of organizing...

BTW, I like the new datastorr release feature.  Worked the first time.

**Comment by richfitz** [comment]
The column specific meta-data is someone else's problem, I think.  Not all the data stored this way will be tabular, in any case.  So as long as there's a facility for including it (most trivially a json file somewhere in the repo that would get slurped in).
